src/Microsoft.DotNet.Darc/src/Darc/Operations/AddBuildToChannelOperation.cs

```diff
@@ -75,6 +87,100 @@ public override async Task<int> ExecuteAsync()
             }
         }
 
+        private async Task<int> PromoteBuildAsync(Build build, Channel targetChannel)
+        {
+            LocalSettings localSettings = LocalSettings.LoadSettingsFile(_options);
+            _options.AzureDevOpsPat = (string.IsNullOrEmpty(_options.AzureDevOpsPat)) ? localSettings.AzureDevOpsToken : _options.AzureDevOpsPat;
```

Comment:
@JohnTortugo I think the reason this is failing is because we don't pass an azure devops token when running the Arcade update test, meaning both sides of the implicit if here result in a null AzDO PAT.

What's the expected behavior of this command in such cases? Probably fail with a readable error message about needing to provide a PAT it in order to use the command?

Reply:
Talked offline.